roxie/ccd/ccdquery.cpp

```diff
@@ -88,7 +93,53 @@ class CQueryDll : implements IQueryDll, public CInterface
     CQueryDll(const char *_dllName, ILoadedDllEntry *_dll) : dllName(_dllName), dll(_dll)
     {
         StringBuffer wuXML;
-        if (!selfTestMode && getEmbeddedWorkUnitXML(dll, wuXML))
+        if (getEmbeddedGraphXML(dll, wuXML))
+        {
+            graph.setown(createPTreeFromXMLString(wuXML));
+            if (!graph->hasProp("node"))
+            {
+                assert(streq(graph->queryName(), "node"));
+                graph->setPropInt("@id", 2);
+                if (!graph->hasProp("hint[@name='dynamic']"))
+                    graph->addPropTree("hint", createPTreeFromXMLString("<hint name='dynamic' value='1'/>"));
+                const char *kindName = graph->queryProp("att[@name='_kind']/@value");
+                ThorActivityKind kind;
+                if (!kindName)
+                    kind = TAKdiskread;
+                else if (strieq(kindName, "DiskRead"))
```

Comment:
Should probably use a function to map names ->kinds

Reply:
There's only likely to be 3 or 4 of the 70-odd that we want to support - so more efficient this way.